libs/pageserver_api/src/models.rs

```diff
@@ -149,30 +151,86 @@ impl std::ops::Deref for TenantCreateRequest {
     }
 }
 
+pub fn deserialize_option_number_from_string<'de, T, D>(
+    deserializer: D,
+) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: FromStr + serde::Deserialize<'de>,
+    <T as FromStr>::Err: Display,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum NumericOrNull<'a, T> {
+        String(String),
+        Str(&'a str),
+        FromStr(T),
+        Null,
+    }
+
+    match NumericOrNull::<T>::deserialize(deserializer)? {
+        NumericOrNull::String(s) => match s.as_str() {
+            "" => Ok(None),
+            _ => T::from_str(&s).map(Some).map_err(serde::de::Error::custom),
+        },
+        NumericOrNull::Str(s) => match s {
+            "" => Ok(None),
+            _ => T::from_str(s).map(Some).map_err(serde::de::Error::custom),
+        },
+        NumericOrNull::FromStr(i) => Ok(Some(i)),
+        NumericOrNull::Null => Ok(None),
+    }
+}
+
 #[derive(Serialize, Deserialize, Debug, Default)]
 pub struct TenantConfig {
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
     pub checkpoint_distance: Option<u64>,
     pub checkpoint_timeout: Option<String>,
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
     pub compaction_target_size: Option<u64>,
     pub compaction_period: Option<String>,
```

Comment:
Was the idea to keep these humantime's as String?

Reply:
Probably yes, this field should accept something like `3s`.